dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.|
+
+```
+{
+    "amount": 100000000,
+    "currency": "XUS",
+    "action": "auth",
+    "valid_until": 74000,
+    "timestamp": 72322,
+}
+```
+### Action Enum
+The following values are allowed for the Action Enum
+* `charge`
+* `auth`
+* `capture`
+
+## PaymentStatusObject
+| Field | Type | Required? | Description |
+|-|-|-|-|
+| status | [Status enum](#status-enum) | Y | Status of the payment|
+| abort_code |[AbortCode enum](#abortcode-enum)| N | In the case of an `abort` status, this field may be used to describe the reason for the abort. Represents the error code of the corresponding error|
+| abort_message | str | N | Additional details about this error.  To be used only when `code` is populated |
+```
+"status": {
+  "status": "abort",
+  "abort_code": "void",
+  "abort_message": "abort message"
+}
+```
+
+### Status Enum
+* `none` - Initial state
+* `ready_for_settlement` - Payment is ready for settlement on chain. This means the actor has performed all required tasks (on its end). Note that the actual settlement is done by the wallet (sender), so for the receiver this status means it is willing to accept the payment.
+* `authorized` - Funds can be locked. This means the actor has performed all required tasks (on its end) and funds can be locked for a future capture. Note that the actual lock is done by the wallet (sender), so for the receiver this status means it is willing to accept the payment. Status is relevant for auth/capture payments only
+* `needs_init_data`- This means the actor requires basic data to initialize the payment. For the receiver, this status means that the payer data is required. For the sender, it means that the merchant data is required
+* `abort` - Payment is cancelled
+
+### Allowed status transitions
+| Current Status | Next Status | Transition Comments |
+|-|-|-|
+| none | needs_init_data |Initial transition. Most common for P2M scenarios in which the actor requires additional data to initialize the payment|
+| needs_init_data | ready_for_settlement | All required tasks were performed and the payment is ready for settlement |
+| needs_init_data | authorized | All required tasks were performed and the funds are ready to be locked |
+| authorized | ready_for_settlement | Funds captured |
+| authorized | abort | Payment was voided |
+| needs_init_data | abort | Payment aborted before initial data was verified |
+| none | abort | Payment aborted before initialization |
+| ready_for_settlement | abort | One party aborted the payment even though the other party already notified it is ready to settle the payment. For example, it may happen if the sender does not approve merchant data provided by the receiver 
```

Comment:
Like this state transition is a bit confusing since it can be explicitly written as

- merchant - waiting_for_settlement
- customer_vasp - settling

but then you can move to abort only if customer_vasp triggers abort or perhaps somehow the merchant can narrowly introduce an abort message into that time frame (though we could argue that an abort at that point is impossible).

Hence settling / waiting_for_settlement could land in an agreeable settled state or aborted.

Reply:
Removed all state transitions from the DIP